parlai/agents/hugging_face/dict.py

```diff
@@ -125,6 +125,13 @@ def override_special_tokens(self, opt):
 
 
 class DialoGPTDictionaryAgent(Gpt2DictionaryAgent):
+    def _define_special_tokens(self, opt):
+        super()._define_special_tokens(opt)
+        if not opt["add_special_tokens"]:
+            # the original pad token '|<endoftext>|' has another usage in global history end token.
+            self.tokenizer.add_special_tokens({"pad_token": DIALOGPT_PAD_TOKEN})
```

Comment:
you could also just make the mask in model.forward do nothing if NULL_IDX == END_IDX, idk.

Reply:
yea that also can solve my problem. I'm just wondering adding pad token here regardless would be closer to fixing a root cause? or changes here would have any corner cases?

Reply:
I mean, even simpler is just set NULL_IDX = -1 and don't add any special tokens. It already says bs=1 is mandatory for `not add_special_tokens`, and all we really need is for the `|<endoftext>|` token to not be masked out.